src/status_im/chat/events/input.cljs

```diff
@@ -0,0 +1,496 @@
+                          :on-requested (fn [jail-response]
+                                          (event-after command-message jail-response))}]
+     {:dispatch [:request-command-data request-data data-type]})))
+
+(register-handler-fx
+ :send-current-message
+ [trim-v]
+ (fn [{{:keys [current-chat-id] :as db} :db} [chat-id]]
+   (let [chat-id      (or chat-id current-chat-id)
+         chat-command (input-model/selected-chat-command db chat-id)
+         seq-command? (get-in chat-command [:command :sequential-params])
+         chat-command (if seq-command?
+                        (let [args (get-in db [:chats chat-id :seq-arguments])]
+                          (assoc chat-command :args args))
+                        (update chat-command :args #(remove str/blank? %)))
+         set-chat-ui-props-event [:set-chat-ui-props {:sending-in-progress? true}]
+         additional-events (if (:command chat-command)
+                             (if (= :complete (input-model/command-completion chat-command))
+                               [[:proceed-command chat-command chat-id]
+                                [:clear-seq-arguments chat-id]]
+                               (let [text (get-in db [:chats chat-id :input-text])]
+                                 [[:set-chat-ui-props {:sending-in-progress? false}]
+                                  (when-not (input-model/text-ends-with-space? text)
+                                    [:set-chat-input-text (str text const/spacing-char)])]))
+                             [[::send-message nil chat-id]])]
+     {:dispatch-n (into [set-chat-ui-props-event]
+                        (remove nil? additional-events))})))
+
+(register-handler-fx
+ ::check-dapp-suggestions
+ [trim-v]
+ (fn [{{:keys [current-account-id] :as db} :db} [chat-id text]]
+   (let [data (get-in db [:local-storage chat-id])]
+     {:chat-fx/call-jail-function {:chat-id    chat-id
+                                   :function   :on-message-input-change
+                                   :parameters {:message text}
+                                   :context    {:data data
+                                                :from current-account-id}}})))
+
+(register-handler-db
+ :clear-seq-arguments
+ [trim-v]
+ (fn [{:keys [current-chat-id chats] :as db} [chat-id]]
+   (let [chat-id (or chat-id current-chat-id)]
+     (-> db
+         (assoc-in [:chats chat-id :seq-arguments] [])
+         (assoc-in [:chats chat-id :seq-argument-input-text] nil)))))
+
+(register-handler-db
+ :update-seq-arguments
+ [trim-v]
+ (fn [{:keys [current-chat-id chats] :as db} [chat-id]]
+   (let [chat-id (or chat-id current-chat-id)
+         text    (get-in chats [chat-id :seq-argument-input-text])]
+     (-> db
+         (update-in [:chats chat-id :seq-arguments] #(into [] (conj % text)))
+         (assoc-in [:chats chat-id :seq-argument-input-text] nil)))))
+
+(register-handler-fx
+ :send-seq-argument
```

Comment:
yes, looking forward to read about your propositions to simplify the chat input handling logic